### DISCUSSION TITLE
Is it possible that frankenphp uses Pingora instead of Caddy?

### DISCUSSION THREAD

**Discussion by sawirricardo** [php/frankenphp#659] [General]
I wonder if it's a possibility

**Comment by nickchomey** [comment]
I can't imagine this would ever be something the devs would even consider. It's an entirely different language, tool etc... Caddy was chosen because it is fast, flexible and extensible, mature, and written in Go, which itself is easy and fast. It is also easy to interface with php because it can use some C libraries. 

I'm sure that a frankenphp-like tool could be made on top of pingora in another project, but I really don't see the point of it now that frankenphp exists. Whatever performance and other improvements it might offer would be negligible in the broader context of most apps that would be served by it.

What problem would you be trying to solve by using pingora instead of caddy?
